_data-prepper/pipelines/configuration/processors/mutate-string.md

```diff
@@ -0,0 +1,256 @@
+---
+layout: default
+title: Mutate string
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# Mutate string processors
+
+You can change the way that a string appears using mutate string processesors. For example, you can use the `uppercase_string` processor to convert a string to uppercase, and you can use the `lowercase_string` processor to convert a string to lowercase. The following is a list of processors that allow you to mutate a string:
+
+* [substitute_string](#substitute_string)
+* [split_string](#split_string)
+* [uppercase_string](#uppercase_string)
+* [lowercase_string](#lowercase_string)
+* [trim_string](#trim_string)
+
+## substitute_string
+
+The `substitute_string` processor matches a key's value against a regular expression (regex) and replaces all returned matches with a replacement string.
+
+### Configuration
+
+You can configure the `substitute_string` processor with the following options.
+
+Option | Required | Description
+:--- | :--- | :---
+`entries` | Yes | A list of entries to add to an event. |
+`source` | Yes | The key to be modified. |
+`from` | Yes | The regular expression (regex) string to be replaced. Special regex characters such as `[` and `]` must be escaped using `\\` when using double quotes and `\` when using single quotes. For more detailed information, see [Class Pattern](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html). |
+`to` | Yes | The string to be substituted for each match of `from`. |
+
+### Usage
+
+To get started, create the following `pipeline.yaml` file: 
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - substitute_string:
+        entries:
+          - source: "message"
+            from: ":"
+            to: "-"
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+Next, create a log file named `logs_json.log`. After that, replace the `path` of the file source in your `pipeline.yaml` file with your file path. For more detailed information, see [Configuring Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/#2-configuring-data-prepper). 
+
+Before you run Data Prepper, the source appears in the following format:
+
+```json
+{"message": "ab:cd:ab:cd"}
+```
+
+After you run Data Prepper, the source is converted to the following format:
+
+```json
+{"message": "ab-cd-ab-cd"}
+```
+
+`from` defines which string is replaced, and `to` defines the string to replace the `from` string. In the preceding example, all instances of the `from` string `"."` are replaced with the `to` string `"-"`, where string `ab:cd:ab:cd` becomes `ab-cd-ab-cd`. If the `from` regex string does not return a match, the key returns without any changes. 
```

Comment:
"`from` defines which string is replaced, and `to` defines the string to replace the `from` string ... all instances of the `from` string `"."` are replaced with the `to` string `"-"`"

This is all redundant information. Both the `to` and `from` keys are described in Configuration above. And then the information in the statement here is the same thing repeated three different ways. There's still important information here, though. Maybe:

"In the preceding example, string `ab:cd:ab:cd` becomes `ab-cd-ab-cd`. If the `from` regex string does not return a match, the key returns without any changes.